frontend/public/actions/ui.ts

```diff
@@ -240,12 +243,24 @@ export const startImpersonate = (kind: string, name: string) => async (dispatch,
   let subprotocols;
   if (kind === 'User') {
     subprotocols = [`Impersonate-User.${encodedName}`];
-  }
-  if (kind === 'Group') {
+  } else if (kind === 'Group') {
     subprotocols = [`Impersonate-Group.${encodedName}`];
+  } else if (kind === 'UserWithGroups' && groups && groups.length > 0) {
+    // User with multiple groups impersonation
+    // Encode user subprotocol
+    subprotocols = [`Impersonate-User.${encodedName}`];
+    // Encode each group as a separate subprotocol
+    groups.forEach((group) => {
+      const encodedGroup = Base64.encode(
+        String.fromCharCode.apply(String, textEncoder.encode(group)),
+      )
+        .replace(/=/g, '_')
+        .replace(/\//g, '-');
```

Comment:
Instead of repeating this logic for the name and group, should we factor this out into a helper?